Data-Structures-Trie.md

```diff
@@ -0,0 +1,129 @@
+# Data Structure Trie
+
+## Introduction to Trie
+
+The word trie is an inflix of the word "re**trie**val", because the trie can find a single word in a dictionary with only a prefix of the word.
+Trie is an efficient data retrieval data structure, using trie, search complexities can be brought to an optimal limit, i.e. length of the string.
+It is a multi-way tree structure useful for storing strings over an alphabet, when we are storing them.
+It has been used to store large dictionaries of English, say, words in spell-checking programs.
+However, the penalty on tries is the storage requirement.
+
+## What is a trie?
+
+A trie is a tree like data structure which stores strings, and helps you find the data associated with that string using the prefix of the string.
+For example, say you plan on building a dictionary to store strings along with their meanings. You must be wondering why can't I simply use a hash table, to get the information.
```

Comment:
Reference hash table : the existing article `Hash-Tables-and-Hashing Functions.md`